config/lib/seed.js

```diff
@@ -92,3 +65,85 @@ function reportError(err) {
   console.log(err);
   console.log();
 }
+
+function performSeedUser(user) {
+
+  var User = mongoose.model('User');
+
+  return new Promise(function(resolve, reject) {
+
+    var seedUser = {
+      username: 'user',
+      password: 'User_Password1!',
+      provider: 'local',
+      email: 'user@localhost.com',
+      firstName: 'User',
+      lastName: 'Local',
+      displayName: 'User Local',
+      roles: ['user']
+    };
+
+    var seedAdmin = {
+      username: 'admin',
+      password: 'Admin_Password1!',
+      provider: 'local',
+      email: 'admin@localhost.com',
+      firstName: 'Admin',
+      lastName: 'Local',
+      displayName: 'Admin Local',
+      roles: ['user', 'admin']
+    };
+
+    var user = null;
+
+    //If production only seed admin if it does not exist
+    if (process.env.NODE_ENV === 'production') {
+      //Add Local Admin
+
+      user = new User(seedAdmin);
+
+      User.generateRandomPassphrase()
```

Comment:
@ilanbiala the desire is understood, I have discussed elaborately with Michael about it.
As far as I see it's not possible to pass multiple arguments to the callback function in a promise thenable, and so we can't just do:
```javascript
generatePassword()
.then(saveUser)
```
because that will pass only the password from the previous promise.
To also pass in the `user` argument we need to either do a closure based call like I did in this original PR or use the function invocation inside saveUser which will look like this:
```javascript
function saveUser(user) {
  return function(password) {
    return new Promise() .. {
  }
}
```
which I find quite ugly.

Is there another way of doing it? otherwise we need to choose one of the above, or refactor the generatePassword method to possibly take the user argument and return it along with the password.